### PR TITLE
feat: adds fedramp-transform command implementation

### LINKED ISSUE #20: Add a feature to use content from an OSCAL SSP to populate the FedRAMP SSP Appendix A template
## Issue description / feature objectives

Create a new command that will use the control implementation content from an OSCAL SSP to populate fields in the FedRAMP SSP Appendix A Template

## Caveats / Assumptions

- Only the control implementation information would be populated starting at page 26
- `This System` data should always be included in the final output
- Control origination and implementation status population might be conditional based on custom YAML headers/properties available

## Completion Criteria

- [ ] A command is added for the transformation
- [ ] A copy of this [template](https://www.fedramp.gov/assets/resources/templates/SSP-Appendix-A-High-FedRAMP-Security-Controls.docx) is populated (list of fields below) and written to an output directory.
    - [ ] Control Origination (#39 )
    - [ ] Parameters (#39)
    - [ ] Implementation Status (#39)
    - [ ] Description (#39 )
    - [ ] Responsible Role (#39)

> The components to include in the output are configurable by title, but by default all components are included (This can be accomplishing using the trestle `ssp-filter` command)

### PR DESCRIPTION
**Pull request by jpower432**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] All commits are signed-off.

## Summary

Adds `fedramp-transform` command to populate information in the SSP Appendix A template

Smaller diffs can be found:
#27 
#29 
#32 
#34 

Blocked by #46 **This should not be merged to develop until 0.3.0 is released**

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle-fedramp)

### CODE REVIEW

**Review by gvauter — CHANGES_REQUESTED** [review]
One suggestion on not instantiating a class instance just to avoid potential unwanted init behavior down the road.

File: trestle_fedramp/core/ssp_reader.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+        """Get the roles by ID."""
+        return {role.id: role.title for role in as_list(self._ssp.metadata.roles)}
+
+    def read_ssp_data(self) -> FedrampControlDict:
+        """Read the ssp from file and return the data for the FedRAMP Template."""
+        control_dict: FedrampControlDict = {}
+
+        for implemented_requirement in as_list(self._ssp.control_implementation.implemented_requirements):
+            control_id = implemented_requirement.control_id
+            label = self._control_labels_by_id.get(control_id, '')
```
[the remark this answers is not in the document]
Comment (jpower432) [reply trestle_fedramp/core/ssp_reader.py:143]: Good catch. This should be `None` and the label would be absent in this case.

**Review by gvauter — APPROVED** [review]
LGTM

**Review by vikas-agarwal76 — APPROVED** [review]
LGTM

**Comment by jpower432** [comment]
Merging this now that 0.3.0 has been released.

**Pull request merged**